lisa/tools/hyperv.py

```diff
@@ -221,20 +237,21 @@ def exists_switch(self, name: str) -> bool:
         )
         return bool(output.strip() != "")
 
-    def delete_switch(self, name: str) -> None:
+    def delete_switch(self, name: str, fail_on_error: bool = True) -> None:
         if self.exists_switch(name):
             self.node.tools[PowerShell].run_cmdlet(
                 f"Remove-VMSwitch -Name {name} -Force",
                 force_run=True,
+                fail_on_error=fail_on_error,
             )
 
-    def create_switch(self, name: str) -> None:
+    def create_switch(self, name: str, switch_type: str = "Internal") -> None:
```

Comment:
Use an enum for Internal/External, I saw it's used multiple places.

Reply:
Fixed